### ISSUE TITLE
[Release] 2.10.0

### ISSUE THREAD

**Issue by jesmrec** [owncloud/android#2453]
### TASKS
 - [x] [COORDINATION] Ping community guys to publish a blog post, at least one week before releasing.
 - [x] [GIT] Create branch release_2.10.0 in owncloud/android from master
 - [x] [GIT] Create branch release_1.0.0 in owncloud/android-library from master 
 - [x] [DEV] Update min version to 19 + remove Jelly Bean workaround
 - [x] [DEV] Update version number and name in build.gradle in android module
 - [x] [DEV] Update version number and name in build.gradle in oc_jb_workaround module 
 - [x] [DIS] Create changelog file (< 500 chars) and add to CHANGELOG.md in owncloud/android 
 - [x] [QA] Design Test plan @jesmrec
 - [X] [QA] Regression Test plan  @jesmrec
- [x] [DOC] Include new changelog on Releases section of github android repo by drafting the new release 2.10.0 
 - [x] [GIT] Create and sign tag oc-android-2.10.0 in HEAD commit of release branch, in owncloud/android 
 - [x] [GIT] Create and sign tag oc-android-library-1.0.0 in HEAD commit of release branch, in  #owncloud/android-library 
 - [x] [GIT] Merge branch release_1.0.0 in owncloud/android-library, into master
 - [x] [GIT] Merge branch release_2.10.0 in owncloud/android, into master 
 - [x] [DIS] Generate test APKs files from branch master in owncloud/android and perform some basic operations.
 - [x] [DIS] Generate final APK files from signed commit in owncloud/android
 - [x] [DIS] Upload & publish release APK and changelog in Play Store
 - [x] [DIS] Update screenshots and store listing, if needed, in Play Store
 - [x] [DIS] Remove OC JB workaround app from Play Store.
 - [X] [FEEDB] Creation of new specific mail addresses for feedback in  otrs mail @michaelstingl 
 - [x] [SPREADTHEWORD] Write a topic in central about the release. https://central.owncloud.org/t/owncloud-2-10-for-android-released/18900
 - [x] [COORDINATION] Write in corresponding channels.
 - [x] [COORDINATION] Write email to release-coordination mailing list.
 - [x] [SPREADTHEWORD] Write a post in the blog @lefherz 
 - [x] [DOC] Update documentation with the new features. @settermjd 
 - [x] Publish app in F-Droid [AUTOMATIC]
 - [x] [GIT] merge master branch into stable, in owncloud/android-library
 - [x] [GIT] merge master branch into stable, in owncloud/android

### BUGS & IMPROVEMENTS:
- [X] #2431 [DONE]
- [X] (1) Upload notification appears twice [FIXED] 
- [X] (2) Lack of notification when deletion is not allowed in shared folders https://github.com/owncloud/android/issues/2453#issuecomment-469754623  [FIXED] 
- [X] (3) Feedback channel changes [FIXED]

**Comment by jesmrec** [comment]
For this release, we can also include https://github.com/owncloud/android/issues/2431

I guess it will carry the removal of the oc_jb_workaround

**Comment by davigonz** [comment]
### (1)
1. Upload file
2. Swipe down to see the notifications bar, two upload notifications appear:
 - Notification with upload progress
 - Meaningless notification with progress that does not move forward

Current: two notifications for uploads
Expected: show only one notification

Pixel 2 v9 server 10.0.10

**Comment by jesmrec** [comment]
### (2)

1. Share a folder which contains some files with another user, with no deletion permission
2. The recipient tries to delete something inside the shared folder

Current: Anything happens
Expected: files are not allowed to be deleted, but snackbar notification is shown

Pixel2 v9

**Comment by jesmrec** [comment]
### (3) 

Set default mail for feedback to `android-app@owncloud.com` instead of `apps@owncloud.com`

Set mail subject as `Android <version> - Feedback` instead of `Feedback - Android <version>`

**Comment by jesmrec** [comment]
All fixed. QA approved.

**Comment by michaelstingl** [comment]
> * [ ]  [DOC] Update documentation with the new features. @settermjd

@settermjd the new features are explained and illustrated with screenshots in the central post:
https://central.owncloud.org/t/owncloud-2-10-for-android-released/18900
/cc @PVince81 FYI ^

**Comment by PVince81** [comment]
where is the mobile documentation hosted ? this is where the ticket about updating it should be filed and scheduled @jesmrec @michaelstingl

**Comment by davigonz** [comment]
> where is the mobile documentation hosted ? this is where the ticket about updating it should be filed and scheduled @jesmrec @michaelstingl

> where is the mobile documentation hosted ? 

Good question, maybe @settermjd can bring us some light.

We used to create documentation issues for each new Android version but https://github.com/owncloud/documentation is now read only, what's the new process now? Using 
https://github.com/owncloud/docs instead? 

![Screenshot 2019-03-14 at 10 51 54](https://user-images.githubusercontent.com/15261363/54347392-3bb74680-4647-11e9-8102-0c01f2087248.png)

**Comment by michaelstingl** [comment]
> where is the mobile documentation hosted ? this is where the ticket about updating it should be filed and scheduled @jesmrec @michaelstingl

Android docs are (here) in the Android repo:
https://github.com/owncloud/android/tree/master/docs/modules/ROOT/pages
(same for other clients)

> https://github.com/owncloud/documentation but now is read only, what's the new process? Using
> https://github.com/owncloud/docs instead?

Yes, exactly.

> This repository is no longer accepting new issues and PRs. If you want to make or suggest a change to the documentation, you need to create an issue or a PR in the new docs repository at: https://github.com/owncloud/docs.

**Comment by PVince81** [comment]
ok, if it's in this repo here then either we reuse this ticket or raise a new ticket for the docs.

in any case, I can assign this to Matthew next week to process the one checkbox item

**Comment by michaelstingl** [comment]
> then either we reuse this ticket or raise a new ticket for the docs.

just let us know what works best for you

**Comment by PVince81** [comment]
@jesmrec can you raise tickets in the docs repository for the new features and also add enough information about the details to be documented ? (don't let @settermjd spend time doing research if you already have enough context to share).

one ticket per feature please.

**Comment by PVince81** [comment]
I don't see any new feature in the changelog above but only bugfixes :confused:

**Comment by davigonz** [comment]
> I don't see any new feature in the changelog above but only bugfixes

We create this issue just for release tasks and bug fixing once the code is frozen. The changelog is in https://github.com/owncloud/android/releases/tag/oc-android-2.10.0

**Comment by jesmrec** [comment]
the current issue is to track the work in the release.

We use to create some documentation stuff after every release, together with @lefherz and other people:

changelog: https://github.com/owncloud/android/blob/master/CHANGELOG.md
blog post: https://owncloud.org/news/owncloud-android-2-10-0-release-thanks-to-the-community/
central.org: https://central.owncloud.org/t/owncloud-2-10-for-android-released/18900

not sure if this is enough. I do not get the target about creating a new issue for every feature, since  we are not performing continuos releasing. We always support @settermjd (or whoever) if further info is needed.

**Comment by PVince81** [comment]
@jesmrec I don't think you can expect @settermjd or anyone who is assigned to write documentation to guess what changelog topics are about. Also don't assume that @settermjd is the only person that would document. I'd prefer if we work in a "push" fashion and people who know most about the feature should write details about it in tickets so that @settermjd only needs to format, structure, clarify, etc.

For example if I look at "Support more options to enforce password when sharing publicly", it doesn't tell me what options are available and doesn't even link back to a ticket or PR where more information should be available (also don't expect people to read the code). Instead of having @settermjd or any doc writer have to do the digging, I'd like you guys to prepare the information to be included in the documentation based on what you already know. This will save a lot of time by avoiding needless research.

On the server side we also work that way.

**Comment by PVince81** [comment]
(now if we had a huge documentation team it might be different and those people can spend days and weeks to go digging for information, but we currently don't have that many people)

**Comment by PVince81** [comment]
if you think that multiple tickets are too much for small additions, a single documentation ticket that contains all details to document for this release would be fine as well.

the key is to prevent people to go digging and asking.
the ticket needs to be actionable and the things to document including details should be clear

**Comment by jesmrec** [comment]
We used to ask @settermjd because he takes care about documentation, but we were always open to explain him (or everyone) about everything we develop. No one complained or asked us for a different way to do it.  Let's take one of your approaches:

After every release, we will open a new issue in https://github.com/owncloud/docs with all the stuff to be included. We will detail everything there. From there, other sources will take the info (for example, @lefherz for the blog). In this way, our work will be posted in one place, where everyone will be able to read and use. From here, @settermjd (or whoever) will format and set in the official docu when they have time. 

Is it ok to get this try?

@michaelstingl @davigonz @abelgardep @hosy @PVince81

**Comment by jesmrec** [comment]
of course, we start to do it with the current 2.10

**Comment by davigonz** [comment]
> After every release, we will open a new issue in https://github.com/owncloud/docs with all the stuff to be included. We will detail everything there. From there, other sources will take the info (for example, @lefherz for the blog). In this way, our work will be posted in one place, where everyone will be able to read and use. From here, @settermjd (or whoever) will format and set in the official docu when they have time.

It works for me though it will increase the estimation points for release issues from now on.

**Comment by PVince81** [comment]
@jesmrec @davigonz I understand that this was the way you worked before. Not sure how effective it was given Matthew's usual workload. Let's shift to the new approach and include the time for writing down the doc notes into your estimates. This is also time that is then saved from Matthew. I believe that having this work done upfront by devs is worth it as the knowledge is already there, while Matthew would need more time (story points) than a dev just to gather ideas.

Thanks!

**Comment by davigonz** [comment]
>  I don't think you can expect @settermjd or anyone who is assigned to write documentation to guess what changelog topics are about.

> For example if I look at "Support more options to enforce password when sharing publicly", it doesn't tell me what options are available and doesn't even link back to a ticket or PR where more information should be available (also don't expect people to read the code)

Just to clarify @PVince81 :

After reading these two comments, it seems you thought we were expecting doc guys to read our code but is not the case. As @jesmrec commented in https://github.com/owncloud/android/issues/2453#issuecomment-473805407 , we always write topics in central with the release details and we considered them as a pretty good starting point to write documentation. 

2.10 topic => https://central.owncloud.org/t/owncloud-2-10-for-android-released/

So I guess Matthew's usual workload was transcribing those features to documentation wording. Make people digging our code for information was never our intention.

To sum up, we will make not only those central topics for end users but the transcription of those topics to documentation wording itself.

**Comment by jesmrec** [comment]
Documentation issue created in docs repository: https://github.com/owncloud/docs/issues/771

We can track the docum update there, so, the current issue is done from my side and can be closed if no one has impediments or extra concerns.
